### ISSUE TITLE
SmallMoleculeComponent.to_openff() does not preserve the molecule protonation state

### ISSUE THREAD

**Issue by LilDojd** [OpenFreeEnergy/gufe#297]
## Expected Behavior
All explicit hydrogens are conserved when converting to OpenFF `Molecule`

## Current Behavior
`OFFMolecule` is constructed directly from `SmallMoleculeComponent._rdkit` with `__init__`:

https://github.com/OpenFreeEnergy/gufe/blob/29f98ec0227ed9f30ea6b84096dc13818a9ac8d3/gufe/components/smallmoleculecomponent.py#L183-L196

This essentially calls <pre>OFFMolecule.from_rdkit(rdmol, allow_undefined_stereo=True, <b>hydrogens_are_explicit=False</b>)</pre> which leads to rdkit adding hydrogens

## Possible Solution
Use `.from_rdkit()` constructor directly with `hydrogens_are_explicit` set to True

## Steps to Reproduce
1. Use sdf in [gdp_correct.sdf.zip](https://github.com/OpenFreeEnergy/gufe/files/14601481/gdp_correct.sdf.zip)

<img src="https://github.com/OpenFreeEnergy/gufe/assets/37330594/f7ed5474-f810-4c33-b5c5-73a9a0b5fc3e" width="50%">

```python
gdp = SmallMoleculeComponent.from_sdf_file("gdp_correct.sdf")
gdp.to_openff().to_file("gdp_ugly.sdf", "sdf")
```

`> gdp_ugly.sdf:`
<img src="https://github.com/OpenFreeEnergy/gufe/assets/37330594/1a45beb3-915a-44c6-b5d8-bca1c2bf19a0" width="50%">

**Comment by LilDojd** [comment]
I know this particular SDF has incorrect bond orders, but I think users' input protonation state should still generally be respected

**Comment by ijpulidos** [comment]
That's a good question, is there a special reason why we don't want hydrogens to be explicit? I know there's an use case for this flag, but I can't think of why it would be needed here.

**Comment by mikemhenry** [comment]
if a `SmallMoleculeComponent` is supposed to be explicit (I think it is), then we do need to add `hydrogens_are_explicit=True`

**Comment by IAlibay** [comment]
Agreed with everyone here - the base assumption for SMCs is that all hydrogens have been explicitly set. We should enforce this with `hydrogens_are_explicit=True`.

**Comment by IAlibay** [comment]
@dotsdl I'm adding this to the v1.3 milestone since @jthorton is seeing cases for this happen in standard PLB systems.

**Comment by dotsdl** [comment]
Sounds good, thanks @IAlibay! Will discuss in today's dev call.

**Comment by IAlibay** [comment]
Do we know if there are any gotchas we need to be aware when using `hydrogens_are_explicit=True`?
I.e. is there a reason why this flag isn't on by default with OpenFF (aside from the likely "hydrogens are usually implicit by default in RDKit")?

Possibly pinging @j-wags here.
